library_registry/admin/routes.py

```diff
@@ -10,59 +14,91 @@
     'admin', __name__,
     template_folder='templates')
 
+# Using an `after_request` callback, we refresh any token that is within 30
+# minutes of expiring. Change the timedeltas to match the needs of your application.
+
+
+@admin.after_request
+def refresh_expiring_jwts(response):
+    try:
+        exp_timestamp = get_jwt()["exp"]
+        now = datetime.now(timezone.utc)
+        target_timestamp = datetime.timestamp(now + timedelta(minutes=30))
+        if target_timestamp > exp_timestamp:
+            access_token = create_access_token(identity=get_jwt_identity())
+            response = make_response(response, 201)
+            set_access_cookies(response, access_token)
+        return response
+    except (RuntimeError, KeyError):
+        # Case where there is not a valid JWT. Just return the original response
+        return response
+
+
 @admin.route('/admin/', strict_slashes=False)
 def admin_view():
     return current_app.library_registry.view_controller()
 
+
 @admin.route('/admin/log_in', methods=["POST"])
+@admin.route('/admin/log_in/<jwt_preferred>')
```

Comment:
Should we consider making this a URL parameter so that other devs clearly understand what they're requesting?

Reply:
Hey Mike...could you clarify here? Do you mean that it's difficult to tell that a Boolean should be passed here? Should we be setting defaults on the routes as optional parameters? Is there a different syntax you had in mind?

Or do you mean not Flask URL parameters but general query URL parameters?

Reply:
Updated to <jwt_cookie_boolean> to be more clear as to what the end point is expecting.